openlibrary/plugins/openlibrary/js/realtime_account_validation.js

```diff
@@ -1,11 +1,13 @@
 export function initRealTimeValidation() {
-    const i18nStrings = JSON.parse(document.querySelector('form[name="signup"]').dataset.i18n);
+    const signupForm = document.querySelector('form[name=signup]');
+    const i18nStrings = JSON.parse(signupForm.dataset.i18n);
 
     if (window.grecaptcha) {
         // Callback that is called when grecaptcha.execute() is successful
         function submitCreateAccountForm() {
-            const signupForm = document.querySelector('form[name=signup]')
-            signupForm.submit()
+            if (signupForm.reportValidity()) {
```

Comment:
Since this is a more critical path section of the code, let's do maybe this:

```suggestion
            if (!signupForm.reportValidity || signupForm.reportValidity()) {
```

So older browsers will still work (although ie11 is the only one covered by this, which we've kind of soft stopped supported) https://caniuse.com/?search=reportValidity